File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int,int>mp;
        mp[0]=1;
        int cnt=0,sum=0;
        for(int i=0;i<nums.size();i++) {
            sum+=nums[i];
            int rem = sum-k;
            if(mp.find(rem)!=mp.end()) {
                cnt+=mp[rem];
            }
            mp[sum]++;
        }
        return cnt;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        prefix = defaultdict(int)
        prefix[0]=1
        sum=0
        cnt=0
        for i in range(len(nums)):
            sum+=nums[i]
            rem = sum-k
            cnt+=prefix[rem]
            prefix[sum]+=1
        return cnt;
        